tests/fullstack-test/mpp/apply_with_late_materialization.test

```diff
@@ -0,0 +1,62 @@
+# Copyright 2023 PingCAP, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# https://github.com/pingcap/tidb/issues/49241
+# Preparation.
+mysql> drop table if exists t;
+mysql> drop table if exists t1;
+mysql> create table t(id int, value int);
+mysql> create table t1(id int, value int);
+mysql> insert into t values(10,5),(9,5),(8,5),(7,5),(6,5),(5,5),(4,5),(3,5),(2,5),(1,5);
+mysql> insert into t1 values(2,5);
+# insert more than 8192 rows to make sure late materialization is enabled.
+mysql> insert into t select * from t; insert into t select * from t; insert into t select * from t; insert into t select * from t;
+mysql> insert into t select * from t; insert into t select * from t; insert into t select * from t; insert into t select * from t;
+mysql> insert into t select * from t; insert into t select * from t; insert into t select * from t;
+mysql> alter table t set tiflash replica 1;
+mysql> alter table t1 set tiflash replica 1;
+mysql> analyze table t;
+
+func> wait_table test t t1
+
+# Test.
+mysql> set @@tidb_allow_mpp=1; set @@tidb_opt_enable_late_materialization=1; select (select t.value from t where t.id = t1.id order by t.value limit 1) xx from t1 order by t1.value limit 5;
++------+
+| xx   |
++------+
+|    5 |
++------+
+mysql> set @@tidb_allow_mpp=1; set @@tidb_opt_enable_late_materialization=0; select (select t.value from t where t.id = t1.id order by t.value limit 1) xx from t1 order by t1.value limit 5;
++------+
+| xx   |
++------+
+|    5 |
++------+
+mysql> set @@tidb_allow_mpp=0; set @@tidb_opt_enable_late_materialization=1; select (select t.value from t where t.id = t1.id order by t.value limit 1) xx from t1 order by t1.value limit 5;
```

Comment:
I think you need to set tidb_allow_tiflash_cop to true if tidb_allow_mpp is set to false.

Reply:
And also add a test for batch cop? You can use `tidb_allow_batch_cop` enforce/disable batch cop

Reply:
done